oeps/architectural-decisions/oep-0065-frontend-composability.rst

```diff
@@ -0,0 +1,492 @@
+
+This approach was abandoned because we still believe that MFE independence is a core need for our platform, and we can't go back to a few monolithic frontends.
+
+Combining MFEs into a monorepo
+==============================
+
+A monorepo would co-locate all of our core MFEs and libraries (``@openedx/paragon``, ``@edx/frontend-platform``, ``@edx/frontend-component-header``, ``@edx/frontend-component-footer``, etc.) in the same repository, but maintain their independent release and deployment cycles. We believe this would help us more readily keep consistent dependency versions across MFEs. But it would also introduce a layer of complexity to our code organization and be a highly invasive way of solving our dependency consistency issues, as we'd have to move all of our core frontend code into a new repository.
+
+Further, it wouldn't solve our consistency problems for anyone working with a non-core MFE or library. We want to create parity between the process for core and non-core repositories to ensure our approach is serving everyone's needs, not just maintainers of core repositories.
+
+We acknowledge that there are benefits here, but believe that it's more work than it's worth, is only a partial solution, and we have less complex options available to us.
+
+Doing Nothing
+=============
+
+We feel that the siloing of micro-frontends, the proliferation of dependencies, the difficulty of extending our platform, and the toil of ongoing maintenance is untenable. This requires us to act to improve the approachability of our frontend architecture; it's not good enough yet.
+
+Appendix A: How Module Federation Works
+***************************************
+
+Without reproducing the details of Webpack's module federation documentation, effectively the following steps occur:
+
+- A host application is provided a list of remote entry points, either through build-time configuration or via an API request to a runtime configuration service (like the MFE config API).
+- Each guest application has a "remote entry" JavaScript file which acts as a manifest of the modules that application exposes, each of which is its own JavaScript file. The guest's versions of all of its dependencies are also present in the manifest.
+- The host application loads those remote entry points as scripts to the document's <head> tag for later use.
+- To load a module, the host application loads the module's JavaScript file based on the URL in the remote entry file, and Webpack's runtime provides that module with any shared dependencies already loaded in the host application.
+- If the Webpack runtime can't reconcile the version of a dependency needed by a guest module with those already loaded, it will use the remote entry manifest to load the guest's version as well.
+
+Change History
```

Comment:
Two notes: (1) Please put the most recent update first; (2) No need to make updates while a PR is under review, because the discussion history can be traced in the PR. See eg https://github.com/openedx/open-edx-proposals/blob/master/oeps/processes/oep-0054-core-contributors.rst change history.

Reply:
Do we want to "squash" the change log entries since the beginning?  I'd been adding them (and duplicating it in the commit messages) so folks could see what changed, but am happy to clean it up.